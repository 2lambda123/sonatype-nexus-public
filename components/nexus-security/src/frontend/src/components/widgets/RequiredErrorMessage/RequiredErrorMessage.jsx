/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import classNames from 'classnames';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faExclamationCircle} from '@fortawesome/free-solid-svg-icons';
import React from 'react';

import './RequiredErrorMessage.scss';

import UIStrings from '../../../constants/UIStrings';

export default function RequiredErrorMessage({className}) {
  const classes = classNames('nxrm-required-error-message', className);
  return <span className={classes}>
  <FontAwesomeIcon icon={faExclamationCircle}/>
  <span className='nxrm-required-error-message-text'> {UIStrings.ERROR.FIELD_REQUIRED} </span>
</span>;
}
