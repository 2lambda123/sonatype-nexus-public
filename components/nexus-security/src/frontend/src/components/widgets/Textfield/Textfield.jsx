/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import './Textfield.scss';

import RequiredErrorMessage from '../RequiredErrorMessage/RequiredErrorMessage';

export default function Textfield({name, value, onChange, isRequired, className}) {
  const isMissingRequiredValue = isRequired && !value;
  const classes = classNames('nxrm-textfield', className, {
    'missing-required-value': isMissingRequiredValue
  });
  return <>
    <input
        name={name}
        type='text'
        value={value}
        onChange={onChange}
        className={classes}
    />
    {isMissingRequiredValue ? <RequiredErrorMessage/> : null}
  </>;
}

Textfield.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  isRequired: PropTypes.bool
};
