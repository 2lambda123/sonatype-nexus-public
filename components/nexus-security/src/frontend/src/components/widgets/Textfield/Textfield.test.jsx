/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {shallow} from 'enzyme';
import React from 'react';

import RequiredErrorMessage from '../RequiredErrorMessage/RequiredErrorMessage';
import Textfield from './Textfield';

describe('Textfield', () => {
  const getTextField = (extraProps) => {
    let props = {
      name: 'testTextfield',
      value: 'val',
      onChange: () => {
      },
      ...extraProps
    };
    return shallow(<Textfield {...props} />);
  };

  it('renders correctly', () => {
    expect(getTextField()).toMatchSnapshot();
  });

  it('hides the error message by default', () => {
    const wrapper = getTextField();

    expect(wrapper.containsMatchingElement(<RequiredErrorMessage/>)).toBe(false);
    expect(wrapper.find('input').hasClass('missing-required-value')).toBe(false);
  });

  it('hides the error message when the value is required and not empty', () => {
    const wrapper = getTextField({
      isRequired: true
    });

    expect(wrapper.containsMatchingElement(<RequiredErrorMessage/>)).toBe(false);
    expect(wrapper.find('input').hasClass('missing-required-value')).toBe(false);
  });

  it('shows the error message when the value is required but empty', () => {
    const wrapper = getTextField({
      isRequired: true,
      value: ''
    });

    expect(wrapper.containsMatchingElement(<RequiredErrorMessage/>)).toBe(true);
    expect(wrapper.find('input').hasClass('missing-required-value')).toBe(true);
  });
});
