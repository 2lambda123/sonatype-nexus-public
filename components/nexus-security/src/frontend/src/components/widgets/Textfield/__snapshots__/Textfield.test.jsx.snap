// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield renders correctly 1`] = `
<Fragment>
  <input
    className="nxrm-textfield"
    name="testTextfield"
    onChange={[Function]}
    type="text"
    value="val"
  />
</Fragment>
`;
